logging/logrus/middleware.go

```diff
@@ -21,12 +21,14 @@ var (
 //
 // All handlers will have a Logrus logger in their context, which can be fetched using `ctxlogrus.Extract`.
 func Middleware(entry *logrus.Entry, opts ...Option) httpwares.Middleware {
+	entry.Logger.Hooks.Add(&timeHook{})
 	return func(nextHandler http.Handler) http.Handler {
 		options := evaluateMiddlewareOpts(opts)
 		return http.HandlerFunc(func(resp http.ResponseWriter, req *http.Request) {
 			wrappedResp := httpwares.WrapResponseWriter(resp)
 
-			requestFields := appendFields(newServerRequestFields(req), options.requestFieldExtractor(req))
+			startTime := time.Now()
+			requestFields := appendFields(newServerRequestFields(req, startTime), options.requestFieldExtractor(req))
```

Comment:
whilst were in could we change this to use `ctxLogrus` the post call uses it but for some reason not pre call.